clouddrift/wavelet.py

```diff
@@ -0,0 +1,801 @@
+"""
+This module provides wavelet functions.
+"""
+
+import numpy as np
+from typing import Optional, Tuple, Union
+import xarray as xr
+import warnings
+from scipy.special import gamma as _gamma, gammaln as _lgamma
+
+
+def morse_wavelet_transform(
+    x: np.ndarray,
+    gamma: float,
+    beta: float,
+    radian_frequency: np.ndarray,
+    complex: Optional[bool] = False,
+    order: Optional[int] = 1,
+    normalization: Optional[str] = "bandpass",
+    boundary: Optional[str] = "mirror",
+    time_axis: Optional[int] = -1,
+) -> Union[Tuple[np.ndarray], np.ndarray]:
+    """
+    Apply a continuous wavelet transform to an input signal using the generalized Morse
+    wavelets of Olhede and Walden (2002). The wavelet transform is normalized differently
```

Comment:
I think we should have the full reference below. See in numpy for example they have ([link](https://github.com/scipy/scipy/blob/v1.11.2/scipy/integrate/_ode.py#L103-L571)):

```
    "dopri5"

        This is an explicit runge-kutta method of order (4)5 due to Dormand &
        Prince (with stepsize control and dense output).

        Authors:

            E. Hairer and G. Wanner
            Universite de Geneve, Dept. de Mathematiques
            CH-1211 Geneve 24, Switzerland
            e-mail:  ernst.hairer@math.unige.ch, gerhard.wanner@math.unige.ch
```